src/Sulu/Bundle/AdminBundle/Resources/js/containers/Datagrid/stores/DatagridStore.js

```diff
@@ -9,30 +9,51 @@ export default class DatagridStore {
     @observable data: Array<Object> = [];
     @observable selections: Array<string | number> = [];
     @observable loading: boolean = true;
+    @observable reset: boolean = false;
+    @observable appendRequestData: boolean = false;
     disposer: () => void;
     resourceKey: string;
     options: Object;
     observableOptions: ObservableOptions;
-    appendRequestData: boolean;
     localeInterceptDisposer: () => void;
-    dataObserveDisposer: () => void;
+
+    static getAppendRequestData: (loadingStrategy: string) => boolean = (loadingStrategy: string) => {
```

Comment:
Why don't we rename the `appendRequestData` to `loadingStrategy`? Then we wouldn't need this method at all, and also setting a new `loadingStrategy` becomes trivial.